pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,35 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional openshift components are specified in the
+// install-config to be included in the installation
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.BaselineCapabilitySet == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("baselineCapabilitySet"), "baseline capability set required"))
+
+		return allErrs
+	}
+	// Check to see if the additionalEnabledCapabilities is defined in openshift/api
+	if availableCapabilities, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; found {
+		// create a set of the available capability set
+		availableCapabilitySet := sets.NewString()
+		for _, capability := range availableCapabilities {
+			availableCapabilitySet.Insert(string(capability))
+		}
+		// create a set of the user specified additional capability set
+		additionalCapabilitySet := sets.NewString()
```

Comment:
Same comment as above. Perhaps something like `userCapabilities` or whatever is consistent with similar checks in the code.